src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java

```diff
@@ -0,0 +1,114 @@
+
+    protected ArrayList<String> getEventNamesToSend() {
+        ArrayList<String> eventNames = new ArrayList<>();
+        eventNames.add("EiffelActivityCanceledEvent");
+        eventNames.add("EiffelActivityStartedEvent");
+        eventNames.add("EiffelActivityFinishedEvent");
+        eventNames.add("EiffelActivityTriggeredEvent");
+        eventNames.add("EiffelAnnouncementPublishedEvent");
+        eventNames.add("EiffelArtifactCreatedEvent");
+        eventNames.add("EiffelArtifactPublishedEvent");
+        eventNames.add("EiffelArtifactReusedEvent");
+        eventNames.add("EiffelCompositionDefinedEvent");
+        eventNames.add("EiffelConfidenceLevelModifiedEvent");
+        eventNames.add("EiffelEnvironmentDefinedEvent");
+        eventNames.add("EiffelFlowContextDefinedEvent");
+        eventNames.add("EiffelIssueVerifiedEvent");
+        eventNames.add("EiffelSourceChangeCreatedEvent");
+        eventNames.add("EiffelSourceChangeSubmittedEvent");
+        eventNames.add("EiffelTestCaseCanceledEvent");
+        eventNames.add("EiffelTestCaseFinishedEvent");
+        eventNames.add("EiffelTestCaseStartedEvent");
+        eventNames.add("EiffelTestCaseTriggeredEvent");
+        eventNames.add("EiffelTestExecutionRecipeCollectionCreatedEvent");
+        eventNames.add("EiffelTestSuiteFinishedEvent");
+        eventNames.add("EiffelTestSuiteStartedEvent");
+        eventNames.add("EiffelArtifactDeployedEvent");
+        eventNames.add("EiffelServiceAllocatedEvent");
+        eventNames.add("EiffelServiceDeployedEvent");
+        eventNames.add("EiffelServiceDiscontinuedEvent");
+        eventNames.add("EiffelServiceReturnedEvent");
+        eventNames.add("EiffelServiceStartedEvent");
+        eventNames.add("EiffelServiceStoppedEvent");
+        eventNames.add("EiffelAlertAcknowledgedEvent");
+        eventNames.add("EiffelAlertCeasedEvent");
+        eventNames.add("EiffelAlertRaisedEvent");
+
+        return eventNames;
+    }
+
+    protected void checkResult() {
+        try {
+            ArrayList<String> eventNames = getEventNamesToSend();
+            String eventsDocument = FileUtils.readFileToString(new File(jsonFilePath), "UTF-8");
+            ObjectMapper objectmapper = new ObjectMapper();
+            JsonNode eventsJson = objectmapper.readTree(eventsDocument);
+
+            for (String temp : eventNames) {
+                String expectedEvent = eventsJson.at("/" + temp).toString();
+                String actualEventID = eventsJson.at("/" + temp + "/meta/id").asText();
+                String document = objectHandler.findObjectById(actualEventID);
+                JsonNode actualJson = objectmapper.readTree(document);
+                JsonNode actualEvent = actualJson.at("/aggregatedObject");
+                ObjectNode objectActualEvent = (ObjectNode) actualEvent;
+                objectActualEvent.without("TemplateName");
+                assertEquals(objectActualEvent.toString(), expectedEvent);
+            }
```

Comment:
Is it possible to get any test report of which events is missing and so on, if it fails.
A test report that says what failed and which event was missing during the failed test execution should really be helpful when debugging.

Reply:
I have created separate story for improving the test result report of what have failed.